lib/routes/authorization.js

```diff
@@ -142,6 +146,9 @@ module.exports = {
         .valid(ACCESS_TYPE_OFFLINE, ACCESS_TYPE_ONLINE)
         .default(ACCESS_TYPE_ONLINE)
         .optional(),
+      // PKCE support
+      code_challenge_method: Joi.string().optional(), // Force 'S256'
```

Comment:
Spec says:

> If the client is capable of using "S256", it MUST use "S256", as
   "S256" is Mandatory To Implement (MTI) on the server.  Clients are
   permitted to use "plain" only if they cannot support "S256" for some
   technical reason and know via out-of-band configuration that the
   server supports "plain".